### PR TITLE
Add support for self-monitoring

### PR DESCRIPTION
**Pull request by zreigz**

### CODE REVIEW

**Comment by zreigz** [comment]
PTAL @matrixik @witekest @Brandstetter

**Comment by Brandstetter** [comment]
1. "Self-monitoring" suggests that the agent monitors itself...I would call it e.g. Monasca-monitoring
2. I would expect a statement about all the other Monasca components which are not mentioned here...seems only kafka, zk and mysql are monitored...

**Comment by Brandstetter** [comment]
ok, I try to explain (we talk about ->monasca-agent-collector/README.md)
2. If we talk about self-monitoring of Monasca and only some components are mentioned the question is: what about the other Monasca components (like log-api or elasticsearch)? Are they monitored? Can they switched off too (MONASCA_MONITORING=true)?
Just a comment would be fine...

3. well, the readme is just for the Monasca agent-collector container for monitoring Monasca. So my comment is not relevant...

**Comment by zreigz** [comment]
Got it, thanks

**Comment by witekest** [comment]
Please remember to [sign your commits](https://github.com/monasca/monasca-docker/blob/master/CONTRIBUTING.md#signing-commits).

**Comment by zreigz** [comment]
PTAL @matrixik @witekest @Brandstetter

**Comment by zreigz** [comment]
PTAL @matrixik @witekest @Brandstetter

**Comment by zreigz** [comment]
---- docker-compose.yml
line 53: the agent is collecting from the same "source" as the cadvisor...Does that mean we have duplicate metrics?

This is required to have access to host file system in order to read `/proc` directory for monasca-agent.
I have compared metrics batween cadvisor and process. The cadvisor returns host specific metrics like a file system, memory, network but monasca-agent with process plugin returns metrics for a specific process like: cpu_perc, io.write_kbytes, io.read_count. 
The answer is the metrics are not duplicated.

**Comment by zreigz** [comment]
line 180, 181: Notification component sends metrics to statsd. Even if MONASCA_MONITORING is set to"false"?

YES - this statsd is independent.

**Comment by zreigz** [comment]
What about log-api?

When MONASCA_LOG_MONITORING is enabled then for log-api the `http check` and 'process' metrics are available.

**Comment by zreigz** [comment]
---- start.sh
line 75: what about:
template $PLUGIN_TEMPLATES/http_check.yaml.j2 $AGENT_PLUGINS/http_check.yaml

template $PLUGIN_TEMPLATES/process.yaml.j2 $AGENT_PLUGINS/process.yaml
?

It's a good question because if someone wants to enable only log monitoring then it is needed.
I have to add those files and test it.

**Comment by zreigz** [comment]
PTAL @matrixik

File: monasca-agent-collector/templates/mysql.yaml.j2 [hunk 1]
```diff
@@ -0,0 +1,8 @@
+init_config:
+instances:
+- built_by: MySQL
+  name: "{{ MYSQL_SERVER | default('mysql') }}"
+  pass: "{{ MYSQL_PASSWORD | default('secretmysql') }}"
```
[the remark this answers is not in the document]
Comment (zreigz) [reply monasca-agent-collector/templates/mysql.yaml.j2]: done

File: monasca-agent-collector/templates/mysql.yaml.j2 [hunk 2]
```diff
@@ -0,0 +1,8 @@
...
+  port: "{{ MYSQL_PORT | default('3306') }}"
```
[the remark this answers is not in the document]
Comment (zreigz) [reply monasca-agent-collector/templates/mysql.yaml.j2]: done

File: monasca-agent-collector/start.sh [hunk 3]
```diff
@@ -61,6 +61,28 @@ if [ "$PROMETHEUS" = "true" ]; then
   template $PLUGIN_TEMPLATES/prometheus.yaml.j2 $AGENT_PLUGINS/prometheus.yaml
 fi
 
+# monasca-monitoring
+if [ "$MONASCA_MONITORING" = "true" ]; then
+  template $PLUGIN_TEMPLATES/zk.yaml.j2 $AGENT_PLUGINS/zk.yaml
+  template $PLUGIN_TEMPLATES/kafka_consumer.yaml.j2 $AGENT_PLUGINS/kafka_consumer.yaml
+  template $PLUGIN_TEMPLATES/mysql.yaml.j2 $AGENT_PLUGINS/mysql.yaml
+  template $PLUGIN_TEMPLATES/http_check.yaml.j2 $AGENT_PLUGINS/http_check.yaml
```
[the remark this answers is not in the document]
Comment (zreigz) [reply monasca-agent-collector/start.sh]: done

File: monasca-agent-collector/templates/zk.yaml.j2 [hunk 4]
```diff
@@ -0,0 +1,6 @@
+init_config:
+
+instances:
+  - host: "{{ ZOOKEEPER_HOST | default('zookeeper') }}"
```
[the remark this answers is not in the document]
Comment (zreigz) [reply monasca-agent-collector/templates/zk.yaml.j2]: No

**Pull request closed without merging**